### PR TITLE
Initial attempt to add support for the fine offset wh0290 air quality monitor

### PR DESCRIPTION
**Pull request by harpchad**
I've tested for 24 hours. There's still a battery indicator field that needs to be identified, but so far it's reliably report values. I tested along side the WH25 and WH24 to ensure the code doesn't affect decoding of their packets. Looks good so far.

### CODE REVIEW

File: src/devices/fineoffset.c [hunk 1]
```diff
@@ -317,6 +317,70 @@ static int fineoffset_WH24_callback(r_device *decoder, bitbuffer_t *bitbuffer)
...
+    aa 2d d4 42 cc 41 9a 41 ae c1 99 9
+             FF DD ?P PP ?A AA ?? CC
+
+- F: 8 bit Family Code?
+- D: 8 bit device id?
+- ?: 2 bits ?
+- P: 14 bit PM2.5 reading in ug/m3
+- ?: 2 bits ?
+- A: 14 bit PM10 reading in ug/m3
+- ?: 8 bits ?
```
Comment (merbanan) [inline src/devices/fineoffset.c]: Should be crc like other fine offset sensors also use.

Comment (harpchad) [reply src/devices/fineoffset.c]: right you are. I pushed another commit that includes the crc8 in addition to the bitsum. That tests out.

File: src/devices/fineoffset.c [hunk 2]
```diff
@@ -317,6 +317,76 @@ static int fineoffset_WH24_callback(r_device *decoder, bitbuffer_t *bitbuffer)
...
+    int pm25          = (b[2] & 0x3f) << 8 | b[3];
+    int pm100         = (b[4] & 0x3f) << 8 | b[5];
+
+
+    /* clang-format off */
+    data = data_make(
+            "model",            "",             DATA_STRING, _X("Fineoffset-WH0290","Fine Offset Electronics, WH0290"),
+            "id",               "ID",           DATA_INT,    id,
+            "25_particles",     "2.5 micron particles",  DATA_FORMAT, "%i ug/m3", DATA_INT, pm25/10,
+            "100_particles",     "10 micron particles",  DATA_FORMAT, "%i ug/m3", DATA_INT, pm100/10,
```
Comment (zuckschwerdt) [inline src/devices/fineoffset.c]: Conventional names are "coarse particulate matter", (PM10) and "fine particulate matter" (PM2.5), maybe we can find a better key based on that. (e.g. should be "pm10_ug_m3" -- the dot in 2.5 could be a problem though…)

Comment (harpchad) [reply src/devices/fineoffset.c]: Maybe pm2_5_ug_m3 and pm10_ug_m3 to avoid the dot?

Comment (harpchad) [reply src/devices/fineoffset.c]: Here's what those look like:
```
time      : @0.084044s
model     : Fine Offset Electronics, WH0290        ID        : 204
2.5um Fine Particulate Matter: 9 ug/m3             10um Coarse Particulate Matter: 10 ug/m3            Integrity : CHECKSUM
```
```
{"time" : "@0.084044s", "model" : "Fine Offset Electronics, WH0290", "id" : 204, "pm2_5_ug_m3" : 9, "pm10_ug_m3" : 10, "mic" : "CHECKSUM"}
```

Comment (harpchad) [reply src/devices/fineoffset.c]: I Pushed the new names and descriptions

Comment (matthewwall) [reply src/devices/fineoffset.c]: fwiw, in the weewx air-quality schema we have been using pm_10_0, pm_2_4, and pm_1_0

Comment (zuckschwerdt) [reply src/devices/fineoffset.c]: Seems a good scheme. Please add the `_0` onto the `10` for uniform looks.

Comment (zuckschwerdt) [reply src/devices/fineoffset.c]: Oh, too late. I'll commit that change directly :)

**Comment by zuckschwerdt** [comment]
Oops, forgot about this. Heads up: I've normalized the `pm10_ug_m3` to `pm10_0_ug_m3` with e9452cf now.

**Pull request merged**